docs/infrastructure/logging.md

```diff
@@ -10,250 +10,272 @@ order: 6
-##### Parameters
+The appenders are defined by the `logging-service.appenders`. Currently only a
+single appender can be selected at a time. The selection may also be done via an
+environmental variable `$ENSO_DEFAULT_APPENDER`.
+
+#### Format
+
+The pattern follows the classic's
+[PatternLayout](https://logback.qos.ch/manual/layouts.html#ClassicPatternLayout)
+format.
+
+Appenders that store/display log events can specify the format of the log
+message via `pattern` field e.g.
 
 ```typescript
-{
-  // Log level associated with the message.
-  level: LogLevel;
-  // Timestamp indicating when the message was sent.
-  time: UTCTime;
-  // An identifier of a log group - the group should indicate which component
-  // the message originated from and any (possibly nested) context.
-  group: String;
-  // The actual log message.
-  message: String;
-  // Optional exception associated with the message.
-  exception?: Exception;
-}
+
+  appenders = [
+    {
+      name = "console"
+      pattern = "[%level{lowercase=true}] [%d{yyyy-MM-dd'T'HH:mm:ssXXX}] [%logger] %msg%n%nopex"
+    }
+    ...
+  ]
 ```
 
-The `exception` field may be omitted if there is no exception associated with
-the message.
-
-In general, the `group` name can be arbitrary, but it is often the quallified
-name of the class that the log message originates from and it is sometimes
-extended with additional nested context, for example:
-
-- `org.enso.launcher.cli.Main`
-- `org.enso.compiler.pass.analyse.AliasAnalysis.analyseType`
-
-### Examples
-
-For example, an error message with an attached exception may look like this (the
-class names are made up):
-
-```json
-{
-  "level": 0,
-  "time": 1600864353151,
-  "group": "org.enso.launcher.Main",
-  "message": "Failed to load a configuration file.",
-  "exception": {
-    "name": "org.enso.componentmanager.config.ConfigurationLoaderFailure",
-    "message": "Configuration file does not exist.",
-    "trace": [
-      {
-        "element": "org.enso.componentmanager.config.ConfigurationLoader.load",
-        "location": "ConfigurationLoader.scala:123"
-      },
-      {
-        "element": "org.enso.launcher.Main",
-        "location": "Main.scala:42"
-      }
-    ],
-    "cause": {
-      "name": "java.io.FileNotFoundException",
-      "message": "config.yaml (No such file or directory)",
-      "trace": []
+#### File Appender
+
+File appender directs all log events to a log file:
+
+```
+  {
+    name = "file"
+    append = <boolean, optional>
+    immediate-flush = <boolean, optional>
+    pattern = <string, optional>
+    rolling-policy {
+      max-file-size = <string, optional>
+      max-history = <int, optional>
+      max-total-size = <string, optional>
     }
   }
-}
 ```
 
-Another example could be an info message (without attached exceptions):
+Rolling policy is a fully optional property of File Appender that would trigger
+automatic log rotation. All properties are optional with some reasonable
+defaults if not specified.
```

Comment:
Can we describe what are these 'some reasonable defaults'? Otherwise, where can I find these defaults?